datamanager/splitLargeZip/split_zip.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+"""
+This script is used to split zip files (DM exports) that are above 1GB in size.
+
+Usage:
+    split_zip.py --path C:\\path\\to\\import\\archive.zip
+
+The result of running the script is multiple zip files that are all below 1GB.
+"""
+import argparse
+import csv
+import os
+import tempfile
+import zipfile
+from dataclasses import dataclass
+from io import TextIOWrapper
+from pathlib import Path
+from posixpath import abspath
+
+ONE_GB = 2 ** 30 - 1
+
+
+@dataclass
+class Configs:
+    schema_path: Path = Path()
+    split_path: Path = Path()
+    start_size: int = 0  # size of schema and split file
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Split a dataset into multiple archives.")
+    parser.add_argument("--path", help="the path to the zip file")
+    args = parser.parse_args()
+    process_zip(Path(args.path))
+
+
+def process_zip(path: Path):
+    with tempfile.TemporaryDirectory() as temp_zip_contents:
+        with zipfile.ZipFile(path, "r") as zip_handle:
+            zip_handle.extractall(temp_zip_contents)
+            top_level_paths = {p.name: p for p in Path(temp_zip_contents).joinpath(path.stem).iterdir()}
+
+            Configs.schema_path = top_level_paths["schema.json"]
+            Configs.start_size += Configs.schema_path.stat().st_size
+
+            Configs.split_path = top_level_paths["split.csv"]
+            Configs.start_size += Configs.split_path.stat().st_size
+
+            split_files(top_level_paths["images"], top_level_paths["latest"], temp_zip_contents)
+
+
+def split_files(images_path: Path, latest_path: Path, folder_contents_path: str):
+    image_names = list()
+    current_size = Configs.start_size
+    archive_number = 1
+    dict_image_paths = {path.name: path for path in list(Path(images_path).iterdir())}
+    dict_latest_paths = {path.stem: path for path in list(Path(latest_path).iterdir())}
+
+    for image_name in dict_image_paths:
+        size = dict_latest_paths[image_name].stat().st_size + dict_latest_paths[image_name].stat().st_size
+        current_size += size
+        if current_size < ONE_GB:
+            image_names.append(image_name)
+        else:
+            write_one_zip(folder_contents_path, image_names, archive_number)
+            archive_number += 1
+            current_size = Configs.start_size + size
+            image_names = list()
+            image_names.append(image_name)
+    write_one_zip(folder_contents_path, image_names, archive_number)  # for the last zip
+
+
+def write_one_zip(folder_contents_path: str, image_names: list, archive_number: int):
+    with zipfile.ZipFile("splitArchive" + str(archive_number) + ".zip", "w") as zip_handle:
+        for root, _, files in os.walk(folder_contents_path):
+            for file in files:
+                for image_name in image_names:
+                    if image_name in file:
```

Comment:
Add a comment here to point out that both the image and related metadata file will be picked up for writing due to the `in` check